earthly/cspell/Earthfile

```diff
@@ -0,0 +1,38 @@
+# cspell UDCs and Containers.
+VERSION 0.7
+
+CSPELL:
+    # Spell checking all docs and code is done with cspell
+    # See: cspell.org
+    COMMAND
+
+    # Where we want to run the `lint` from.
+    ARG --required src
+
+    # Unlikely to need to change this.
+    ARG cfg_file=cspell.json
+
+    RUN --no-cache if [[ ! -f $src/$cfg_file ]]; then \
```

Comment:
I don't think the `--no-cache` flags are required here. Think of it like this:

```
COPY src src
DO +CSPELL --src=src
```

In the above case, if the contents of the `src` directory change, then the cache breaks at that layer and the `CSPELL` UDC will always be run. If the contents of `src` _do not_ change, then normally the `CSPELL` UDC wouldn't run. However, the explicit `--no-cache` flags may override that behavior and cause this UDC to always run. 

The net effect is that doing this may _always_ bust the cache when this UDC is used and cause downstream layers to constantly rebuild.

Reply:
There are two contexts here.
1. Its run in CI on the Checked out code of a PR, before any containers start being built.
2. Its run on the local machine, checking the developer's code in their repo.
In both cases it's running on code which is not inside any container being built.
So Earthly has no way to know if the code changed or not.  Better for it to always run the test than falsely say nothing changed.

Reply:
There is no layer.

Reply:
Sure, but this is a result of doing it locally instead of copying the files into the container. If I call this UDC in `TargetA` and then include that target in `TargetB`, then `TargetB` will never get cached because of the usage of `--no-cache` here. 

It seems like not using `LOCALLY` solves this issue.

Reply:
Why would we be linting inside the container?  We shouldn't even build the container if the lints fail?
The whole point of this is to check things are good before we do more complex stuff, like compile code.

Reply:
It seems torturous also to force that to do any listing we must build a container just to hold the code we want to lint.
All I want to do is check does the code on my disk pass before I commit it to the repo, or that I just checked out during a PR action pass.

Its not something we would ever use when making a container.

Reply:
Let's give it a shot. It would be helpful to get a working example with all the pieces in place. Then we can do some tests, especially around how things interact with `--no-cache`.